lib/mapAdCards.test.js

```diff
@@ -0,0 +1,126 @@
+import { determineAdCards } from './mapAdCards'
+import { defaultDisplayFlags } from '../contents/BenefitAdvertisingCards'
+import { StatusCodes } from '../constants/StatusCodes'
+import { ProgramCodes } from '../constants/ProgramCodes'
+import { TypeCodes } from '../constants/ProgramTypeCodes'
+
+describe('Mapping Ad Cards', () => {
+  function generateBenefit(programCode, statusCode, typeCode) {
+    return {
+      programCode: programCode,
+      statusCode: statusCode,
+      typeCode: typeCode,
+      summaries: [],
+    }
+  }
+
+  let adCards
+  beforeEach(() => {
+    //Deep copy the default flags to reset flags each test run
+    adCards = { ...defaultDisplayFlags }
+  })
+
+  it('EI - Status inactive sets EI flag to false', () => {
+    const eiBenefit = generateBenefit(
+      ProgramCodes.EI,
+      StatusCodes.inactive,
+      TypeCodes.EISickness
+    )
+    determineAdCards(eiBenefit, adCards)
+    expect(adCards.EI).toEqual(false)
+  })
+
+  it('EI - All statuses except inactive keeps EI flag as true', () => {
+    Object.entries(StatusCodes).forEach(([statusCode, statusString]) => {
+      if (statusCode !== 'inactive' || statusString !== StatusCodes.inactive) {
```

Comment:
there is no reset of adCards, so this function may not always work as expected. I would suggest separating this into multiple tests. A test should test one thing at a time; single input, single output.